File: ios/Sources/Detonator/Request/OpenUrlRequest.swift
import Foundation
import UIKit

class OpenUrlRequest: Request {
    override public func run() {
        let urlString: String! = decodeData()
        
        guard let url = URL(string: urlString) else {
            error(message: "Not a valid url.")
            
            return
        }
        
        if !UIApplication.shared.canOpenURL(url) {
            error(message: "Can't open url.")
            
            return
        }
        
        UIApplication.shared.open(url, options: [:], completionHandler: nil)
        
        end()
    }
}

File: android/Detonator/src/main/java/com/iconshot/detonator/request/OpenUrlRequest.java
package com.iconshot.detonator.request;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;

import com.iconshot.detonator.Detonator;

public class OpenUrlRequest extends Request<String> {
    public OpenUrlRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    @Override
    public void run() {
        String url = decodeData(String.class);

        try {
            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));

            detonator.context.startActivity(browserIntent);

            end();
        } catch (ActivityNotFoundException e) {
            error(new Exception("Can't open url."));
        }
    }
}
